lib/utils/getSelectorAST.mjs

```diff
@@ -0,0 +1,19 @@
+import selectorParser from 'postcss-selector-parser';
+
+/**
+ * @param {string} selector
+ * @param {import('stylelint').PostcssResult} result
+ * @param {import('postcss').Node} node
+ * @returns {import('postcss-selector-parser').Root | undefined}
+ */
+export default function getSelectorAST(selector, result, node) {
+	if (!selector) return undefined;
+
+	try {
+		return selectorParser().astSync(selector);
+	} catch (err) {
+		result.warn(`Cannot parse selector (${err})`, { node, stylelintType: 'parseError' });
```

Comment:
nit

I would expect it to be in `err.message`.

Reply:
This function mirrors `lib/utils/parseSelector.mjs`.
I'd prefer to keep this as is for now.

Maybe we should consider changing all instances of `catch (err) {}`?
But that is beyond the scope of this change.